### PR TITLE
TRN-1: Add vpc construct

### LINKED ISSUE #1: Setup CDKTF Constructs for AWS VPC and Budget Alert Configuration #1
## Overview

This issue focuses on establishing the foundational AWS infrastructure using CDKTF (Cloud Development Kit for Terraform). The primary components to be configured are the Virtual Private Cloud (VPC) with specific subnet setups.

### Objectives

- [x] Implement a VPC with a detailed subnet configuration.

#### VPC Configuration
- [x] Create a VPC with the following specifications:
  - 3 private subnets.
  - 3 public subnets.
- [x] Configure route tables for each subnet.
- [x] Set up NAT gateways for private subnets.
- [x] Establish associations between subnets and route tables.
- [ ] Ensure the VPC configuration aligns with AWS best practices.

### Dependencies
- CDKTF setup and configuration.
- AWS Account with necessary permissions.

### References
- [CDKTF Documentation](https://www.terraform.io/cdktf)
- [AWS VPC Documentation](https://docs.aws.amazon.com/vpc/latest/userguide/what-is-amazon-vpc.html)
- [AWS Budgets Documentation](https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/budgets-managing-costs.html)

### Additional Notes
- Security and scalability should be considered in the VPC setup.

### PR DESCRIPTION
**Pull request by grok-rs**

### CODE REVIEW

File: lib/aws/vpc/index.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        subnetId: subnet.id,
+      });
+    });
+
+    const privateRouteTableAssociations = privateRouteTables.map((privateRouteTable, index) => {
+      return new aws.routeTableAssociation.RouteTableAssociation(this, `route-table-private-association-${index}`, {
+        routeTableId: privateRouteTable.id,
+        subnetId: privateSubnets[index].id,
+      });
+    });
```
Comment (atre) [inline lib/aws/vpc/index.ts]: These both are declared but never used. Please run `compile` or `watch` commands to catch stuff like this

**Review by atre — CHANGES_REQUESTED** [review]
Please fix all code issues and let's sync on linting before merging this

File: lib/aws/vpc/index.ts (hunk 1 above)
Comment (grok-rs) [reply lib/aws/vpc/index.ts]: removed

File: lib/aws/vpc/index.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
+import * as aws from '@cdktf/provider-aws';
```
Comment (atre) [inline lib/aws/vpc/index.ts]: Please import each class separately, e.g. `import { Vpc } from '@cdktf/provider-aws/lib/vpc'` and change all the usages

File: lib/aws/vpc/index.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import { DEFAULT_CIDR } from './constant';
+import { SubnetOptions, SubnetType, VpcOptions } from './interface';
+
+export class VpcConstruct extends Construct {
+  constructor(scope: Construct, id: string, private opts: VpcOptions = {}) {
+    super(scope, id);
+
+    const vpc = new aws.vpc.Vpc(this, 'vpc',
+      {
+        tags: { Name: opts?.name ?? '' },
```
Comment (atre) [inline lib/aws/vpc/index.ts]: Please use objects destructuring for opts and remove to props please

**Review by atre — CHANGES_REQUESTED** [review]
Please fix import and destructuring issues

File: package.json [hunk 4]
```diff
@@ -1,13 +1,16 @@
 {
   "name": "trn-components",
-  "version": "0.1.0",
+  "version": "0.2.0",
```
Comment (atre) [inline package.json:3]: Please change to 0.1.0

File: lib/aws/vpc/interface.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  cidr?: string;
+  azs?: string[];
+  publicSubnets?: string[];
+  privateSubnets?: string[];
+  enableNatGateway?: boolean;
+  singleNatGateway?: boolean;
+  enableDnsHostnames?: boolean;
+  tags?: { [key: string]: string };
+  publicSubnetTags?: { [key: string]: string };
+  privateSubnetTags?: { [key: string]: string };
```
Comment (atre) [inline lib/aws/vpc/interface.ts:12]: Please change to Record<>

File: lib/aws/remote-backend/index.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export { RemoteBackendConstruct } from './remote-backend';
```
Comment (atre) [inline lib/aws/remote-backend/index.ts:1]: Please move all related code to index

**Review by atre — APPROVED** [review]
Please make changes as were discussed in a call

**Pull request merged**